From mithlesh@linsyssoft.com  Fri Mar 20 19:56:06 2009
From: Mithlesh Thukral <mithlesh@linsyssoft.com>
Date: Fri, 20 Mar 2009 17:35:12 +0530 (IST)
Subject: Staging: sxg: Fix interface bug not working after open is called second time.
To: Greg Kroah-Hartman <gregkh@suse.de>
Cc: Sahara Project <saharaproj@linsyssoft.com>, Christopher Harrer <charrer@alacritech.com>
Message-ID: <alpine.LFD.2.00.0903201734440.11454@localhost.localdomain>


Fix the problem of the interface not working after a sequence of up-down-up
events. The problem was observed only on systems where the card was using MSI-X
interrupts. On the second time open, the driver did not request MSI-X vector
but was trying to use normal interrupts.

Signed-off-by: LinSysSoft Sahara Team <saharaproj@linsyssoft.com>
Signed-off-by: Mithlesh Thukral <mithlesh@linsyssoft.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/sxg/sxg.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/staging/sxg/sxg.c
+++ b/drivers/staging/sxg/sxg.c
@@ -2254,7 +2254,7 @@ int sxg_second_open(struct net_device * 
         SXG_ENABLE_ALL_INTERRUPTS(adapter);
 
 	netif_carrier_on(dev);
-        sxg_register_interrupt(adapter);
+	sxg_register_intr(adapter);
 	spin_unlock_irqrestore(&sxg_global.driver_lock, sxg_global.flags);
 	return (STATUS_SUCCESS);
 
